### DISCUSSION TITLE
Guidance for conditionally loading endpoints

### DISCUSSION THREAD

**Discussion by hknlof** [quarkusio/quarkus#27088] [Q&A]
We would like to enable `@Path` annotated endpoints conditionally. For e.g. a config `QUARKUS_AUTH_ENABLED` is set, then this would enable the Endpoint `/authentication` otherwise a `404 Not Found` should be returned.

Is there way to UNLOAD these classes from the `ClassLoader`? How would you approach a potential solution?

**Comment by stephan-strate** [comment]
You could just implement this logic in the endpoint directly.
Read the config, if enabled process with the business logic, if disabled return another status code.

**Reply by hknlof** [reply]
Unfortunately, that would mean a lot of Endpoints need to have this Code enabled. Most of our Endpoints need to be checked against certain criteria for their availability. The approach I am now taking is using extensions and following the pattern of [smallrye-metrics](https://github.com/quarkusio/quarkus/tree/main/extensions/smallrye-metrics)

**Reply by stephan-strate** [reply]
You could also use an interceptor and implement the config logic once.

**Reply by hknlof** [reply]
Yepp, unfortunately the sources of some endpoints need to reside in a different folder / sourceSet. I could not get Gradle sourceSets playing nicely with quarkusDev and quarkusBuild.
